src/Form/SsofactRegisterForm.php

```diff
@@ -74,22 +88,22 @@ public function buildForm(array $form, FormStateInterface $form_state) {
     if (!$client_config->get('enabled')) {
       return $form;
     }
-    // Use custom redirect_uri for client-side login form, which does not check
-    // state token.
+
     $redirect_uri = Url::fromRoute('ssofact.redirect_login', [
       'client_name' => 'ssofact',
     ], [
       'absolute' => TRUE,
       'language' => \Drupal::languageManager()->getLanguage(LanguageInterface::LANGCODE_NOT_APPLICABLE),
       'query' => [
-        'target' => Url::fromUri('internal:/')->toString(),
+        'destination' => Url::fromRoute('<current>')->toString(),
```

Comment:
The parameter `target` is a required parameter of the ssoFACT API authorization endpoint and should be the final redirect URI path on the client application.  We should not change this parameter name, as we don't know whether that will be supported in future versions of ssoFACT.

After discussing the addition of an alter hook in [OpenIDConnectRedirectController](https://cgit.drupalcode.org/openid_connect/tree/src/Controller/OpenIDConnectRedirectController.php?h=8.x-1.x#n132) to avoid having to customize the value of `$_SESSION['openid_connect_destination']` with you, I remember that you asked whether it is okay to instead patch the openid_connect module to add support for a custom `destination` parameter.  However, we only ever discussed to change the inbound query parameter handling inside Drupal, not the parameter name that we pass externally to ssoFACT.  We have total control over the inbound parameters, but the outbound parameters are part of their defined interfaces.  Sorry if this distinction was not clear enough.

https://github.com/netzstrategen/drupal-ssofact/commit/bc9103ca08ec7da43c3417e6d8b07c6ee507bbf7 has been added for the Talk integration today, which performs the mapping from `target` to `destination` for all inbound requests, and which should also eliminate the need for specifying a different parameter in all the cases here.

I'm not sure whether it also eliminates the need for the openid_connect patch though…  I guess there can still be an override value for `destination` in the user session, if it was stored (together with a state) upfront?

IIRC, that was the starting point for having to manipulate the destination parameter in the first place, so it is probably still required.  This exact override situation could also be clarified some more in https://www.drupal.org/project/openid_connect/issues/2988428

Reply:
Ok, thanks for clarify, I will provide a commit fixing the to instances of s/target/destination substitution. :+1: